### DISCUSSION TITLE
Adaptive layout on mobile devices for Superset Dashboard

### DISCUSSION THREAD

**Discussion by memes-forever** [apache/superset#20734] [Show and tell]
## Screenshot

![image](https://user-images.githubusercontent.com/107195976/179173509-b7a94c34-2472-4bc4-b414-b19f377704ed.png)

## Description

Let us continue our discussion on this issue, as it is required by many users.

## Past Topics
- https://github.com/apache/superset/issues/13725
- https://github.com/apache/superset/discussions/13816
- https://github.com/apache/superset/issues/6472

## Design
I suggest stretching the blocks to the width of the screen, placing the blocks in a column.

![image](https://user-images.githubusercontent.com/107195976/179173132-a6c4c869-f17b-4168-92b6-035172d0b5b8.png)

## Style example:
```
@media only screen and (max-width: 790px) {
  .grid-row.background--transparent {
    flex-wrap: wrap;
  }
  .dragdroppable-column{
    min-width: 95%;
    max-width: 95%;
    margin-bottom: 16px;
  }
  .resizable-container{
    width: 100%;
    max-width: 100%;
  }
}
```

**Comment by srinify** [comment]
Nice! Moving to Discussion as this isn't a bug or "issue" :)

**Comment by BoogalooLi** [comment]
thanks for your example. But in the table Top 10 Games, the data cannot be shown properly on mobile device. Do you have some better attempts?

**Comment by coderyeasin** [comment]
I also try it out but not working in mobile device or tab or laptop size. this media query its not applied properly in dashboard. how can i resolve it anyone have idea? please share it with us!

**Comment by lelelethu** [comment]
Is there anyone have solution for that?

**Comment by samsonitto** [comment]
I also struggle with this problem. Did anyone find any solution?

**Comment by CoderSxy** [comment]
try this
` @media screen and (max-width: 700px) {
      .grid-container {
        margin-left: 10px !important;
        margin-right: 10px !important;
      }
      .grid-row {
        flex-wrap: wrap;
        .dragdroppable-column {
          width: 100%;
          margin-right: 0 !important;
          margin-bottom: 20px;
          &:last-child {
            margin-bottom: 0;
          }
          .resizable-container {
            width: 100% !important;
            max-width: 100% !important;
          }
        }
      }
    }`
    
   And also need to be in the src/dashboard/components/gridComponents/ChartHolder TSX added a judgment
   `if (isFullSize) {
      chartWidth = window.innerWidth - CHART_MARGIN;
      chartHeight = window.innerHeight - CHART_MARGIN;
    } else if (
      window.innerWidth < 700 &&
      document.querySelector('.slice_container')?.clientWidth > 100
    ) {
      chartWidth = document.querySelector('.slice_container')?.clientWidth;
      chartHeight = Math.floor(
        component.meta.height * GRID_BASE_UNIT - CHART_MARGIN,
      );
    } else {`
